### ISSUE TITLE
Styles not being removed on live-reload

### ISSUE THREAD

**Issue by marshallswain** [donejs/donejs#47]
While using steal-tools live-reload with the below template (from the guide), if I remove the line `<can-import from="place-my-order-assets" />` the styles don't get removed until I manually refresh.

```mustache
<html>
  <head>
    <title>{{title}}</title>
    {{asset "css"}}
  </head>
  <body>
    <can-import from="place-my-order-assets" />
    <can-import from="place-my-order/styles.less!" />
    <can-import from="place-my-order/app" as="viewModel" />

    <h1>{{message}}</h1>

    {{asset "inline-cache"}}

    {{#switch @env.NODE_ENV}}
      {{#case "production"}}
        <script src="/node_modules/steal/steal.production.js"  main="place-my-order/index.stache!done-autorender"></script>
      {{/case}}
      {{#default}}
        <script src="/node_modules/steal/steal.js"></script>
      {{/default}}
    {{/switch}}
  </body>
</html>
```

**Comment by matthewp** [comment]
## Orphaned modules

This problem has to deal with orphaned modules.  Let's say you have a module **A** that depends on **B** and **C**.  If A reloads and now only depends on B, C is an orphan, it doesn't get removed from the page like it should.

### Solution

1. Currently the server only sends the string name of the module that changed like **A**.  The first solution I can think of is to have the server instead send an object like `{name:"A", deps:["B"]}` and this way we know to delete children that are not part of the current deps.

The downside is that this would be a breaking change.

2. We could also use the parentMap in the client to detect orphans. This might be expensive since we'll have to look at every module in the map to detect this... would be nice if we had a childMap.

**Comment by matthewp** [comment]
I think I'm going to go with solution 2, maintain a childMap in live-reload so that when modules are reloaded we will know what modules changed and delete the orphans.

**Comment by matthewp** [comment]
Changed my mind, 1 is the way to go. We can't reliably know what a module's dependencies are in the client. So we'll send a message from the server that includes modules that were removed.

**Comment by justinbmeyer** [comment]
> We can't reliably know what a module's dependencies are in the client. 

Why?

**Comment by matthewp** [comment]
Fixed by #73
